docs/installation/install-with-configuration-file.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Deploying Karmada Using Configuration File with CLI
+---
+
+This documentation serves as a comprehensive guide for deploying Karmada using a configuration file through the `karmadactl` CLI tool.
+
+## Prerequisites
+
+- A Kubernetes cluster ready for Karmada installation.
+
+- A valid kube-config file to access your Kubernetes cluster.
+
+- The karmadactl command-line tool or kubectl-karmada plugin installed.
+
+  :::note
+
+  For installing `karmadactl` command line tool refer to the [Installation of CLI Tools](/docs/next/installation/install-cli-tools#one-click-installation) guide.
+
+  :::
+
+## Installation
+
+`karmadactl init` allows you to install Karmada by specifying a configuration file, which provides a structured way to define all settings in a YAML file.
+
+### Example Configuration File
+Here is an example of the configuration file for Karmada deployment:
+```yaml
+apiVersion: config.karmada.io/v1alpha1
+kind: KarmadaInitConfig
+metadata:
+  name: karmada-init
+spec:
+  karmadaCrds: "https://github.com/karmada-io/karmada/releases/download/v1.10.3/crds.tar.gz"
+  etcd:
+    local:
+      imageRepository: "registry.k8s.io/etcd"
+      imageTag: "3.5.13-0"
+      initImage:
+        imageRepository: "docker.io/library/alpine"
+        imageTag: "3.19.1"
+  components:
+    karmadaAPIServer:
+      imageRepository: "registry.k8s.io/kube-apiserver"
+      imageTag: "v1.30.0"
+    karmadaAggregatedAPIServer:
+      imageRepository: "docker.io/karmada/karmada-aggregated-apiserver"
+      imageTag: "v1.10.3"
+    kubeControllerManager:
+      imageRepository: "registry.k8s.io/kube-controller-manager"
+      imageTag: "v1.30.0"
+    karmadaControllerManager:
+      imageRepository: "docker.io/karmada/karmada-controller-manager"
+      imageTag: "v1.10.3"
+    karmadaScheduler:
+      imageRepository: "docker.io/karmada/karmada-scheduler"
+      imageTag: "v1.10.3"
+    karmadaWebhook:
+      imageRepository: "docker.io/karmada/karmada-webhook"
+      imageTag: "v1.10.3"
+```
+
+### Deploying Karmada with Configuration File
+
+1.Save the example configuration above to a file, e.g., karmada-config.yaml.
+
+2.Use the following command to deploy Karmada with the configuration file:
+
+  ```bash
+  sudo karmadactl init --karmada-apiserver-replicas 3 --etcd-replicas 3 --etcd-storage-mode PVC --storage-classes-name <storage-classes-name> --kubeconfig <path-to-kube-config>
```

Comment:
I can't see the `--config` here.

By the way, are these flags mandatory? `--karmada-apiserver-replicas 3 --etcd-replicas 3 --etcd-storage-mode PVC --storage-classes-name <storage-classes-name>`
I think we are demonstrating how to use the `--config` flag, it would be great to focus on that.

Reply:
> I can't see the `--config` here.
> 
> By the way, are these flags mandatory? `--karmada-apiserver-replicas 3 --etcd-replicas 3 --etcd-storage-mode PVC --storage-classes-name <storage-classes-name>` I think we are demonstrating how to use the `--config` flag, it would be great to focus on that.

Yes, this was a mistake. We should be focusing on config-related commands, and when I used the `ha-installation-with-cli.md` file as a template, I mistakenly copied the content. I have now updated it with the correct command examples.